app.js

```diff
@@ -1,129 +1,143 @@
+const bookHolder = localStorageHandler.getBookHolder();
+let inputData = localStorageHandler.getInputData();
+function addBookToHolder(newBook) {
+  // Check if the book already exists in local storage
+  if (bookHolder.some((book) => book.title === newBook.title)) {
+    // Book already exists, do nothing
+    return;
+  }
+  bookHolder.push(newBook);
+  localStorageHandler.saveBookHolder(bookHolder);
+  displayBooks();
+}
 
-  addButton.addEventListener('click', () => {
-    if (inputTitle.value === '' || inputAuthor.value === '') {
-      bookDisplay.append(error);
-    } else {
-      addBook();
-    }
-  });
-});
+function clearInputData() {
+  inputData = { inputTitle: '', inputAuthor: '' };
+  localStorageHandler.saveInputData(inputData);
+}
+function addBook() {
+  const titleData = domElements.inputTitle.value;
+  const authorData = domElements.inputAuthor.value;
+  if (titleData === '' || authorData === '') {
+    domElements.displayError();
+    return;
+  }
+  const newBook = new Book(titleData, authorData);
+  addBookToHolder(newBook);
+  clearInputData();
+  displayBooks();
+  domElements.inputTitle.value = '';
+  domElements.inputAuthor.value = '';
+}
 
-// Creating a new book instance and appending it to the bookDisplay div
-function display() {
-  bookDisplay.innerHTML = ''; // Clearing the previous instances before displaying the updated ones
+function displayBooks() {
+  domElements.bookDisplay.innerHTML = '';
   bookHolder.forEach((book, index) => {
     const bookInstance = document.createElement('article');
-    const dispTitle = document.createElement('h2');
-    dispTitle.textContent = book.title;
-    const dispAuthor = document.createElement('p');
-    dispAuthor.textContent = book.author;
+    const dispTitle = `"${book.title}" by ${book.author}`;// Concatenate book title and author
     const delButton = document.createElement('button');
-    delButton.textContent = 'remove';
+    delButton.textContent = 'Remove';
     delButton.setAttribute('data-index', index);
     delButton.addEventListener('click', deleteBook);
-    const hrline = document.createElement('hr');
-    bookInstance.append(dispTitle, dispAuthor, delButton, hrline);
-    bookDisplay.append(bookInstance);
+
+    bookInstance.append(
+      document.createTextNode(dispTitle),
+      delButton,
+    );
+
+    domElements.bookDisplay.append(bookInstance);
+
+    // add class to each bookInstance element based on its index
+    if (index % 2 === 0) {
+      bookInstance.classList.add('book-row-even');
+    } else {
+      bookInstance.classList.add('book-row-odd');
+    }
   });
 }
-// Removing a book from the bookHolder array and updating localStorage and the UI
+
 function deleteBook(event) {
   const index = event.target.getAttribute('data-index');
   bookHolder.splice(index, 1);
-  localStorage.setItem('storedBooks', JSON.stringify(bookHolder));
-  display();
+  localStorageHandler.saveBookHolder(bookHolder);
+  displayBooks();
```

Comment:
- [ ] As requested by the previous reviewer, Please make sure the delete book function is class-level and implemented inside the class to follow the project requirements.
  - [Optional] I would suggest making all the functions class-level.